module/VuFind/tests/unit-tests/src/VuFindTest/RecordDriver/SolrDefaultTest.php

```diff
@@ -208,15 +208,97 @@ public function testGetHighlightedTitle()
     }
 
     /**
-     * Test getHighlightedSnippet for a record.
+     * Test getHighlightedSnippet for an empty record.
      *
      * @return void
      */
-    public function testGetHighlightedSnippet()
+    public function testEmptyGetHighlightedSnippet()
     {
         $this->assertEquals(false, $this->getDriver()->getHighlightedSnippet());
     }
 
+    /**
+     * Test getHighlightedSnippet for a record when empty snippet data is given.
+     *
+     * @return void
+     */
+    public function testGetHighlightedSnippetAllEmpty()
+    {
+        $overrides = [
+            'General' => ['snippets' => true],
+        ];
+        $driver = $this->getDriver([], $overrides);
+        $details = ['topic' => ['']];
+        $driver->setHighlightDetails($details);
+        $this->assertEquals(false, $driver->getHighlightedSnippet());
+    }
+
+    /**
+     * Test getHighlightedSnippet for a record when the first snippet is empty.
+     *
+     * @return void
+     */
+    public function testGetHighlightedSnippetFirstEmpty()
+    {
+        $overrides = [
+            'General' => ['snippets' => true],
+        ];
+        $driver = $this->getDriver([], $overrides);
+        // Note that the first topic result is empty, it should return the first non-empty one
+        $details = ['topic' => ['', 'Testing {{{{START_HILITE}}}}Snippets{{{{END_HILITE}}}} highlighting']];
+        $driver->setHighlightDetails($details);
+        $this->assertEquals(
+            ['snippet' => 'Testing {{{{START_HILITE}}}}Snippets{{{{END_HILITE}}}} highlighting','caption' => false],
```

Comment:
Minor point of style: in all of the arrays in the test, there is no space after the comma before 'caption'.